test/integration/acceptance/grant_permission_test.cpp

```diff
@@ -0,0 +1,884 @@
+#include "builders/protobuf/transaction.hpp"
+#include "builders/protobuf/queries.hpp"
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "framework/specified_visitor.hpp"
+#include "validators/permissions.hpp"
+
+using namespace integration_framework;
+using namespace shared_model;
+
+namespace shared_model {
+  namespace proto {
+    class QueryResponse;
+  }
+}
+
+class GrantPermissionTest : public AcceptanceFixture {
+ public:
+
+  /**
+   * Creates the transaction with the user creation commands
+   * @param perms are the permissions of the user
+   * @return built tx and a hash of its payload
+   */
+  auto makeAccountWithPerms(const std::string &user,
+                            const crypto::Keypair &key,
+                            const std::vector<std::string> &perms,
+                            const std::string &role) {
+    return createUserWithPerms(user, key.publicKey(), role, perms)
+        .build()
+        .signAndAddSignature(kAdminKeypair)
+        .finish();
+  }
+
+  /**
+   * Forms a transaction such that creator of transaction
+   * grants permittee a permission
+   * @param creatorAccountName — a name, e.g. user
+   * @param permitteeAccountId — an id of grantee, mostlikely name@test
+   * @param grantPermission — any permission from the set of grantable
+   * @return a Transaction object
+   */
+  proto::Transaction accountGrantToAccount(const std::string &creator_account_name,
+                                           const crypto::Keypair &creator_key,
+                                           const std::string &permitee_account_name,
+                                           const std::string &grant_permission) {
+    const std::string creator_account_id = creator_account_name + "@" + kDomain;
+    const std::string
+        permitee_account_id = permitee_account_name + "@" + kDomain;
+    return TestUnsignedTransactionBuilder()
+        .creatorAccountId(creator_account_id)
+        .createdTime(iroha::time::now())
+        .grantPermission(permitee_account_id, grant_permission)
+        .quorum(1)
+        .build()
+        .signAndAddSignature(creator_key)
+        .finish();
+  }
+
+  /**
+   * Forms a transaction that either adds or removes signatory of an account
+   * @param f Add or Remove signatory function
+   * @param permitee_account_name
+   * @param permitee_key
+   * @param account_name
+   * @return a transaction
+   */
+  proto::Transaction permiteeModifySignatory
+      (TestUnsignedTransactionBuilder (TestUnsignedTransactionBuilder::*f)(
+          const interface::types::AccountIdType &,
+          const interface::types::PubkeyType &) const,
+       const std::string &permitee_account_name,
+       const crypto::Keypair &permitee_key,
+       const std::string &account_name) {
+    const std::string
+        permitee_account_id = permitee_account_name + "@" + kDomain;
+    const std::string
+        account_id = account_name + "@" + kDomain;
```

Comment:
Is it formatted?

Reply:
![screen shot 2018-06-11 at 20 26 33](https://user-images.githubusercontent.com/11841667/41247132-bffa6528-6db5-11e8-9538-206d1bfc50ef.png)

Reply:
After clang-format mine looks like that
![screenshot_20180611_220805](https://user-images.githubusercontent.com/5658339/41251869-f99f8426-6dc3-11e8-8e03-ddaa4b46fc57.png)